### PR TITLE
Strimzi Performance Testing 

### PR DESCRIPTION
**Pull request by see-quick**

### CODE REVIEW

File: 068-performance-testing.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+1. **Uphold High Performance Standards:**
+    - Performance testing ensures that Strimzi not only meets but exceeds the performance standards expected by users for managing Kafka clusters in diverse environments.
+2. **Enable Continuous Improvement:**
+    - With dedicated performance benchmarks, Strimzi can continually assess and enhance its components, driving innovation while maintaining performance integrity.
+3. **Build Confidence Among Users:**
+    - By demonstrating a commitment to performance excellence, Strimzi strengthens trust with its user community, reassuring them that their Kafka infrastructures are managed with utmost efficiency.
+
+Moreover, we can integrate these performance tests into our Azure pipelines, enabling automatic checks to identify
+if a PR could cause a significant performance degradation, thereby ensuring every change is vetted for efficiency before being merged (future vision).
+This integration represents a proactive step towards maintaining high-performance standards and continuous improvement in Strimzi's development process.
```
Comment (fvaleri) [inline 068-performance-testing.md]: This sounds like a repetition of point 2.

File: 068-performance-testing.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+There are a benefits if we introduce performance testing within the Strimzi such as:
+1. **Uphold High Performance Standards:**
+    - Performance testing ensures that Strimzi not only meets but exceeds the performance standards expected by users for managing Kafka clusters in diverse environments.
+2. **Enable Continuous Improvement:**
+    - With dedicated performance benchmarks, Strimzi can continually assess and enhance its components, driving innovation while maintaining performance integrity.
+3. **Build Confidence Among Users:**
+    - By demonstrating a commitment to performance excellence, Strimzi strengthens trust with its user community, reassuring them that their Kafka infrastructures are managed with utmost efficiency.
+
+Moreover, we can integrate these performance tests into our Azure pipelines, enabling automatic checks to identify
```
Comment (fvaleri) [inline 068-performance-testing.md]: Point 4?

File: 068-performance-testing.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+## Motivation for Performance testing in Strimzi
+
+In the world of distributed systems, performance is not merely a feature but a fundamental aspect that dictates the
+reliability, scalability, and efficiency of the system.
+For Apache Kafka clusters managed by Strimzi, where data throughput, latency, and resource utilization are critical,
+ensuring optimal performance is paramount.
+As enterprises rely on Kafka for real-time data processing and streaming analytics, the performance of the Strimzi operators
+directly influences the operational capabilities and service quality experienced by end-users.
+
+There are a benefits if we introduce performance testing within the Strimzi such as:
```
Comment (fvaleri) [inline 068-performance-testing.md]:
```suggestion
Benefits of performance tests:
```

File: 068-performance-testing.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
+# Strimzi Performance Testing
```
Comment (fvaleri) [inline 068-performance-testing.md]:
```suggestion
# Introducing Performance Testing
```

File: 068-performance-testing.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+## Motivation for Performance testing in Strimzi
```
Comment (fvaleri) [inline 068-performance-testing.md]:
```suggestion
## Motivation
```

File: 068-performance-testing.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+2. **Enable Continuous Improvement:**
+    - With dedicated performance benchmarks, Strimzi can continually assess and enhance its components, driving innovation while maintaining performance integrity.
+3. **Build Confidence Among Users:**
+    - By demonstrating a commitment to performance excellence, Strimzi strengthens trust with its user community, reassuring them that their Kafka infrastructures are managed with utmost efficiency.
+
+Moreover, we can integrate these performance tests into our Azure pipelines, enabling automatic checks to identify
+if a PR could cause a significant performance degradation, thereby ensuring every change is vetted for efficiency before being merged (future vision).
+This integration represents a proactive step towards maintaining high-performance standards and continuous improvement in Strimzi's development process.
+
+There are several considerations for designing such testing, each with its own set of advantages and disadvantages, which are discussed in the subsequent section.
```
Comment (fvaleri) [inline 068-performance-testing.md]:
```suggestion
There are several design considerations, each with its own set of advantages and disadvantages, which are discussed in the following sections.
```

File: 068-performance-testing.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+---
+
+This approach may appear more effective than the initial one, yet it necessitates the creation of numerous support classes for:
+
+1. Extracting metrics from Kubernetes Pods components
+2. Monitoring the entire lifecycle
+3. Documenting (i.e., reporting) the outcomes of performance tests
+    4. this could all metrics scraped during performance test
+    5. and then reporting via table of each parameter monitored during such test
```
Comment (fvaleri) [inline 068-performance-testing.md]:
```suggestion
4. this could all metrics scraped during performance test
5. and then reporting via table of each parameter monitored during such test
```

File: 068-performance-testing.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+For instance such result from tool could look like this:
```
Comment (fvaleri) [inline 068-performance-testing.md]: Which tool?

**Review by fvaleri — COMMENTED** [review]
This is a good start, but I think the proposal should pick one strategy, which is what the author believes it's the best approach, and describe the others in the rejected alternative section, with motivations. That way, you can really think through pros and cons, and take a position. We would also need more details on the output and its format.

Looking at what Kafka does regarding performance tests, I see them side by side with system tests, leveraging the same framework. I think this is a lot easier to maintain, and every project has its own set of system tests. We just need to isolate them in a specific category. We also don't need performance test for all components or use cases, but only for the most critical ones. My 2cents.

**Comment by see-quick** [comment]
> but I think the proposal should pick one strategy, which is what the author believe it's the best approach, and describe the others in the rejected alternative section, with motivations. That way, you can really think through pros and cons, and take a position. We would also need more details on the output and its format.

Yeah, that's what I was thinking when I ended this writing... :D I did not know what to pick because everything seems to have some cons and pros....and letting more people to decide is better (as for this initial draft...), Basically I was 50/50 on having:
1. integrate it into a strimzi-kafka-operator repo
2. having a dedicated repo

So hearing some input from you would be good to consider all the consequences.

File: 068-performance-testing.md (hunk 8 above)
Comment (see-quick) [reply 068-performance-testing.md]: Kinda confusing one....
```
For instance, such result from the performance framework could look like this:
```

**Comment by see-quick** [comment]
Let's hear @im-konge opinion on this...currently, I am not sure which approach would be the best :) But I am not against having system-tests together with performance tests.

**Comment by im-konge** [comment]
I agree with @fvaleri about having the performance tests side-by-side with system tests. The main problem in having the separate repo is dependency on particular component/operator. So in the repository with performance tests you would need to include operators, OAuth, bridge, possibly Drain cleaner (in case that you would like to see how many Kafka/ZK Pods it will handle when the Node will be drained), but you will be limited with the fact that you will not be able to run performance tests for the PRs, RCs etc - that means in case of API change or whatever, you would need to "bump" the dependency.

So in case that you would like to go with the 2nd approach, you will need to keep in mind that you will use the SNAPSHOT versions (most probably) and only when the PRs are merged. Having first approach, you can easily trigger it for each PR and you'll see how much the performance went down after the changes.

I know that I said that having separate PR for performance tests would be clearer, but I thought about it more and now I see that it would not be maybe that beneficial.

Also, I think you can have another module for the performance tests inside the `strimzi-kafka-operator` repository, which can take the things from `systemtest` module and add more needed for the performance tests (or add it to `test-frame` into separate module just for the performance testing, which would be FMPOV also beneficial). The performance tests could be then triggered using profile or just by specifying the package (as we anyway do for the `systemtest` module).

**Comment by see-quick** [comment]
> I agree with @fvaleri about having the performance tests side-by-side with system tests. The main problem in having the separate repo is dependency on particular component/operator. So in the repository with performance tests you would need to include operators, OAuth, bridge, possibly Drain cleaner (in case that you would like to see how many Kafka/ZK Pods it will handle when the Node will be drained), but you will be limited with the fact that you will not be able to run performance tests for the PRs, RCs etc - that means in case of API change or whatever, you would need to "bump" the dependency.
> 
> So in case that you would like to go with the 2nd approach, you will need to keep in mind that you will use the SNAPSHOT versions (most probably) and only when the PRs are merged. Having first approach, you can easily trigger it for each PR and you'll see how much the performance went down after the changes.
> 
> I know that I said that having separate PR for performance tests would be clearer, but I thought about it more and now I see that it would not be maybe that beneficial.
> 
> Also, I think you can have another module for the performance tests inside the `strimzi-kafka-operator` repository, which can take the things from `systemtest` module and add more needed for the performance tests (or add it to `test-frame` into separate module just for the performance testing, which would be FMPOV also beneficial). The performance tests could be then triggered using profile or just by specifying the package (as we anyway do for the `systemtest` module).

Yeah agree, during this weekend or the following week I would re-design a proposal and write more insights on how we want to do that (or at least point on implementation possibles). Thanks

**Comment by see-quick** [comment]
Update, there are a few leftovers but I think it's kinda of ready WDYT @fvaleri @im-konge? Also, there are some questions which I have but... Moreover, maybe I am overcomplicating some things in the design so feel free to be a critic :D

File: 068-performance-testing.md [hunk 9]
```diff
@@ -0,0 +1,487 @@
+#  Introducing Performance Testing
+
+## Motivation
+
+In the world of distributed systems, performance is not merely a feature but a fundamental aspect that dictates the
+reliability, scalability, and efficiency of the system.
```
Comment (fvaleri) [inline 068-performance-testing.md]:
```suggestion
In the world of distributed systems, performance is one of the most important non-functional requirements, which has an impact on system design, reliability, scalability, and efficiency.
```

File: 068-performance-testing.md [hunk 10]
```diff
@@ -0,0 +1,487 @@
...
+
+Benefits of performance tests:
+1. **Uphold High Performance Standards:**
+    - Performance testing ensures that Strimzi not only meets but exceeds the performance standards expected by users for managing Kafka clusters in diverse environments.
+2. **Enable Continuous Improvement:**
+    - With dedicated performance benchmarks, Strimzi can continually assess and enhance its components, driving innovation while maintaining performance integrity.
+3. **Build Confidence Among Users:**
+    - By demonstrating a commitment to performance excellence, Strimzi strengthens trust with its user community, reassuring them that their Kafka infrastructures are managed with utmost efficiency.
+4. **Performance checks (future vision):** 
+    - Moreover, we can integrate these performance tests into our Azure pipelines, enabling automatic checks to identify if a PR could cause a significant performance degradation, thereby ensuring every change is vetted for efficiency before being merged.
```
Comment (fvaleri) [inline 068-performance-testing.md]: I would leave the CI pipeline out of the motivation section, as the benefits can be leveraged even without a CI pipeline, although less efficiently. That said you should mention it later in the implementation approach.

File: 068-performance-testing.md [hunk 11]
```diff
@@ -0,0 +1,487 @@
+#  Introducing Performance Testing
+
+## Motivation
+
+In the world of distributed systems, performance is not merely a feature but a fundamental aspect that dictates the
+reliability, scalability, and efficiency of the system.
+For Apache Kafka clusters managed by Strimzi, where data throughput, latency, and resource utilization are critical,
+ensuring optimal performance is paramount.
```
Comment (fvaleri) [inline 068-performance-testing.md]: In general, you should use one line for each phrase, as this seems to be the convention for proposals. I'm leaving this comment here, but it applies to the entire proposal.

File: 068-performance-testing.md [hunk 12]
```diff
@@ -0,0 +1,487 @@
...
+
+I suggest a few possible methods for integrating performance testing into Strimzi, each characterized by its unique benefits and drawbacks.
+Our decision on which path to follow should align with our specific requirements and goals.
+
+### Approach 1: Integration within the strimzi-kafka-operator STs Module
+
+This method involves incorporating performance testing directly into the existing system test module of our strimzi-kafka-operator repository.
+It's a relatively straightforward strategy, leveraging the wealth of methods, resources, and components already available
+within the repository, such as ResourceManager, LogCollector, and MetricsCollector.
+This integration significantly lowers the complexity and duplication of effort compared to setting up a separate dedicated repository for performance tests.
```
Comment (fvaleri) [inline 068-performance-testing.md]:
```suggestion
This approach significantly lowers the complexity and avoids code duplication compared to setting up a separate dedicated repository for performance tests.
```

File: 068-performance-testing.md [hunk 13]
```diff
@@ -0,0 +1,487 @@
...
+One of the drawbacks that we would need to add a new classes like PerformanceReporting and tools for generating Markdown documents, among others.
```
Comment (fvaleri) [inline 068-performance-testing.md]:
```suggestion
One of the drawbacks is that we would need to add a new classes like PerformanceReporting and tools for generating Markdown documents, among others.
```

File: 068-performance-testing.md [hunk 14]
```diff
@@ -0,0 +1,487 @@
...
+
+## Implementation details
+
+As mentioned above we will have an set of test cases, which could be run by specification of the concrete test profile.
+Execution will be handled by JUnit5 identically as we are using it in `systemtests` module.
+
+### Test case example
+
+As an illustrative example, we can do the following:
+1. create an set of possible configuration for SUT.
```
Comment (fvaleri) [inline 068-performance-testing.md]: I know what this acronym means, but some reader may not be familiar.

File: 068-performance-testing.md [hunk 15]
```diff
@@ -0,0 +1,487 @@
...
+                .withComponentName(this.testStorage.getClusterName())
+                .build();
+
+        this.topicOperatorPollingThread = new TopicOperatorPollingThread(this.topicOperatorCollector, "strimzi.io/cluster=" + this.testStorage.getClusterName());
+        this.daemonThread = new Thread(this.topicOperatorPollingThread);
+        this.daemonThread.setDaemon(true); // Set as daemon so it doesn't prevent JVM shutdown
+        this.daemonThread.start();
+
+        // 2a. Exercise: Creation of KafkaTopics
+        startTime = System.currentTimeMillis();
```
Comment (fvaleri) [inline 068-performance-testing.md]: Let's use nanoTime, that is not subject to clock drifts.

File: 068-performance-testing.md [hunk 16]
```diff
@@ -0,0 +1,487 @@
...
+with both pros and cons.
+One advantage is the potential for rapid adoption and integration of new performance testing methodologies,
+given k6's established user base and continuous updates.
+However, a possible downside is the need to maintain compatibility with Strimzi's evolving infrastructure,
+possibly requiring frequent updates to any Strimzi-specific extensions or configurations within k6.
+
+## Selected Approach/Proposal: Approach 1 - Integration within the strimzi-kafka-operator
+
+Within this framework, we propose two separate pathways to achieve our performance testing integration, 
+each with its unique set of advantages:
```
Comment (fvaleri) [inline 068-performance-testing.md]: Which path do you propose to take?

**Review by fvaleri — COMMENTED** [review]
Thanks. The content is good, but I think you should follow the official template and not have any branch in the proposal description. 

I mean, you should always propose one approach. If you considered alternate approaches, and you think they are relevant for the discussion, please move them to the rejected alternatives section.

In the end, we would need to vote on a single approach.

File: 068-performance-testing.md (hunk 16 above)
Comment (see-quick) [reply 068-performance-testing.md]: integrate it directly to `systemtest` module and the let's see what others think

File: 068-performance-testing.md [hunk 17]
```diff
@@ -0,0 +1,451 @@
...
+####  Notes on Release Cycle for Using k6:
+
+The release cycle for incorporating k6 into Strimzi testing would bear similarities to the dedicated repository approach,
+with both pros and cons.
+One advantage is the potential for rapid adoption and integration of new performance testing methodologies,
+given k6's established user base and continuous updates.
+However, a possible downside is the need to maintain compatibility with Strimzi's evolving infrastructure,
+possibly requiring frequent updates to any Strimzi-specific extensions or configurations within k6.
+
+## Selected Approach/Proposal: Approach 1 - Integration within the strimzi-kafka-operator
```
Comment (im-konge) [inline 068-performance-testing.md]: I would keep just this selected approach and others leave to rejected alternatives
In case that others will agree on different approach, you will have it as one of the commits, so you can just easily replace it 🤷

File: 068-performance-testing.md [hunk 18]
```diff
@@ -0,0 +1,451 @@
...
+
+We propose integrating performance testing directly alongside existing system tests, leveraging the established infrastructure and testing utilities. 
+This method ensures that performance and system tests can share resources efficiently, streamlines the development process, facilitates immediate feedback on performance impacts, and reduces overhead.
+
+- This method integrates performance testing directly alongside existing system tests. 
+- It leverages the already established infrastructure and testing utilities, ensuring that performance and system tests can share resources such as the ResourceManager, LogCollector, and MetricsCollector.
```
Comment (im-konge) [inline 068-performance-testing.md]: These two things are already mentioned on the previous two lines or?

File: 068-performance-testing.md [hunk 19]
```diff
@@ -0,0 +1,451 @@
...
+  - Reduces the overhead of maintaining separate modules for system and performance tests.
+
+### Core parts
+
+The following strategies will be core to our implementation:
+
+1. **JUnit5 Usage:** 
+   1. Employ JUnit5 for structuring and executing performance tests. This choice allows us to benefit from JUnit5's advanced features such as dynamic tests, parameterized tests, and its powerful extension model, enhancing our test design and flexibility.
+2. **Resource Management:** 
+   1. Utilize the ResourceManager from the system test module to manage the lifecycle of resources.
```
Comment (im-konge) [inline 068-performance-testing.md]:
```suggestion
   1. Utilize the `ResourceManager` from the system test module to manage the lifecycle of resources.
```

File: 068-performance-testing.md [hunk 20]
```diff
@@ -0,0 +1,451 @@
...
+   2. This approach ensures efficient setup, teardown, and management of Kafka clusters, topics, and other necessary components across both performance and system tests.
+3. **Metrics Collection:** 
+   1. Base our metrics collection on the system test module's MetricsCollector, with plans to possibly extend this functionality in the future from an external library like Test Frame.
```
Comment (im-konge) [inline 068-performance-testing.md]:
```suggestion
   1. Base our metrics collection on the system test module's `MetricsCollector`, with plans to possibly extend this functionality in the future from an external library like Test Frame. 
```
+ link to test-frame possibly?

File: 068-performance-testing.md [hunk 21]
```diff
@@ -0,0 +1,451 @@
...
+The following strategies will be core to our implementation:
+
+1. **JUnit5 Usage:** 
+   1. Employ JUnit5 for structuring and executing performance tests. This choice allows us to benefit from JUnit5's advanced features such as dynamic tests, parameterized tests, and its powerful extension model, enhancing our test design and flexibility.
+2. **Resource Management:** 
+   1. Utilize the ResourceManager from the system test module to manage the lifecycle of resources.
+   2. This approach ensures efficient setup, teardown, and management of Kafka clusters, topics, and other necessary components across both performance and system tests.
+3. **Metrics Collection:** 
+   1. Base our metrics collection on the system test module's MetricsCollector, with plans to possibly extend this functionality in the future from an external library like Test Frame. 
+   2. Additionally, introduce specialized collectors such as TopicOperatorMetricsCollector to gather detailed metrics relevant to the Strimzi Topic Operator, tailoring our data gathering to the nuances of Kafka and Strimzi.
```
Comment (im-konge) [inline 068-performance-testing.md]:
```suggestion
   2. Additionally, introduce specialized collectors such as `TopicOperatorMetricsCollector` to gather detailed metrics relevant to the Strimzi Topic Operator, tailoring our data gathering to the nuances of Kafka and Strimzi.
```

File: 068-performance-testing.md [hunk 22]
```diff
@@ -0,0 +1,451 @@
...
+   2. The reporter should be capable of generating detailed summaries and insights, 
+   ideally supporting formats conducive to both human reading and automated parsing, such as Markdown or structured JSON.
+
+**Including within the `systemtest` module:** Integrate performance tests to ensure they coexist seamlessly with system tests, potentially categorizing tests to maintain clarity and prevent any confusion between system and performance testing efforts (i.e., using performance profile). 
+
+### CI: Azure Pipelines
+
+We assume Azure machines may not suffice for all performance tests, meaning we anticipate that lower-complexity performance tests, 
+not requiring as much memory/CPU, could run on such infrastructure. However, for tests needing more memory in the context of 
+performance, we prefer another approach (i.e., using a testing farm) to execute these tests.
```
Comment (im-konge) [inline 068-performance-testing.md]:
```suggestion
performance, we prefer another approach (i.e., using a Testing Farm) to execute these tests.
```

File: 068-performance-testing.md [hunk 23]
```diff
@@ -0,0 +1,451 @@
...
+
+### CI: Possible integration with Testing farm
```
Comment (im-konge) [inline 068-performance-testing.md]:
```suggestion
### CI: Possible integration with Testing Farm 
```

File: 068-performance-testing.md [hunk 24]
```diff
@@ -0,0 +1,451 @@
...
+
+### CI: Azure Pipelines
+
+We assume Azure machines may not suffice for all performance tests, meaning we anticipate that lower-complexity performance tests, 
+not requiring as much memory/CPU, could run on such infrastructure. However, for tests needing more memory in the context of 
+performance, we prefer another approach (i.e., using a testing farm) to execute these tests.
+
+### CI: Possible integration with Testing farm 
+
+Special consideration will be given to the potential integration of performance tests with testing farms, utilizing
```
Comment (im-konge) [inline 068-performance-testing.md]: the `testing farms` here seems weird
if you are writing about Testing Farm, then it should have the name in the text
otherwise, if you think of something else, you should name it here I think

File: 068-performance-testing.md [hunk 25]
```diff
@@ -0,0 +1,451 @@
...
+    } finally {
+        // 4a. teardown: Gracefully stop the topic-operator thread
+        // 4b: teardown: Now, it's safe to log performance data as the collection thread has been stopped
+        PerformanceReporter.logPerformanceData(
+                numberOfTopics, numberOfClientInstances, NUMBER_OF_MESSAGES, createTopicsTime, totalSendAndRecvTime, 
+                totalDeletionTime, totalTimeWhole, this.testStorage, ACTUAL_TIME, Environment.PERFORMANCE_DIR, 
+                maxBatchSize, maxBatchLingerMs, this.topicOperatorPollingThread.getMetricsHistory()
+        );
+    }
+}
```
Comment (im-konge) [inline 068-performance-testing.md]: I think pasting the whole testcase here is not necessary
Because it's anyway similar code as we have today in STs and everything inside it is just implementation of a test case.

File: 068-performance-testing.md [hunk 26]
```diff
@@ -0,0 +1,451 @@
...
+```
+
+Note that this is just an one possible example. There is no need to use @ParametrizedTest.
+We could also use simple @Test to edge performance cases.
+
+### Gathered metrics
+
+During test execution we need to scrape metrics from the `SUT` (i.e., system under test) pods.
```
Comment (im-konge) [inline 068-performance-testing.md]: You already explained what SUT is, so you don't need it here again

File: 068-performance-testing.md [hunk 27]
```diff
@@ -0,0 +1,451 @@
...
+}
+```
+
+Note that this is just an one possible example. There is no need to use @ParametrizedTest.
+We could also use simple @Test to edge performance cases.
+
+### Gathered metrics
+
+During test execution we need to scrape metrics from the `SUT` (i.e., system under test) pods. 
+Therefore, one way how to tackle such problem is that we could use a Thread, which would
```
Comment (im-konge) [inline 068-performance-testing.md]:
```suggestion
Therefore, one way how to tackle such problem is that we could use a `Thread`, which would
```

File: 068-performance-testing.md [hunk 28]
```diff
@@ -0,0 +1,451 @@
...
+        } catch (Exception e) {
+            LOGGER.error("Exception in thread", e);
+        } finally {
+            LOGGER.info("Thread finishing");
+        }
+    }
+}
+```
+
+And in the test case we would simply interrupt such Thread to finish with scraping.
```
Comment (im-konge) [inline 068-performance-testing.md]:
```suggestion
And in the test case we would simply interrupt such `Thread` to finish with scraping.
```

File: 068-performance-testing.md [hunk 29]
```diff
@@ -0,0 +1,451 @@
...
+            ├── strimzi_update_status_duration_seconds_max.txt
+            ├── strimzi_update_status_duration_seconds_sum.txt
+            ├── system_cpu_count.txt
+            ├── system_cpu_usage.txt
+            ├── system_load_average_1m.txt
+            ├── system_load_average_per_core_percent.txt
+            └── test-performance-metrics.txt
+... # more dates
+```
+where each of these files contains values, scraped by Thread in the test.
```
Comment (im-konge) [inline 068-performance-testing.md]:
```suggestion
where each of these files contains values, scraped by `Thread` in the test. 
```

File: 068-performance-testing.md [hunk 30]
```diff
@@ -0,0 +1,451 @@
...
+            ├── strimzi_resources.txt
+            ├── strimzi_total_time_spend_on_uto_event_queue_duration_seconds.txt
+            ├── strimzi_update_status_duration_seconds_max.txt
+            ├── strimzi_update_status_duration_seconds_sum.txt
+            ├── system_cpu_count.txt
+            ├── system_cpu_usage.txt
+            ├── system_load_average_1m.txt
+            ├── system_load_average_per_core_percent.txt
+            └── test-performance-metrics.txt
+... # more dates
```
Comment (im-konge) [inline 068-performance-testing.md]: This can be shorter as well I think (I suppose that no-one will actually read it whole)

File: 068-performance-testing.md [hunk 31]
```diff
@@ -0,0 +1,451 @@
...
+development and maintenance efforts. The need to recreate foundational components and ensure their integration with each 
+Strimzi project adds complexity and resource demands.
+3. **Adoption of external performnace tools (e.g., Grafana k6)**:
+Despite its powerful features and extensibility, this option introduces a steep learning curve due to the requirement for proficiency in Go/Javascript, 
+which may not align with the skill set of the existing Strimzi community. 
+Additionally, it necessitates the development and maintenance of Strimzi-specific extensions, complicating the testing process.
+
+## Conclusion
+
+Integrating performance testing within the strimzi-kafka-operator's system test module offers a streamlined, efficient approach to ensuring the performance standards of Strimzi-managed Kafka clusters are met and exceeded.
```
Comment (im-konge) [inline 068-performance-testing.md]:
```suggestion
Integrating performance testing within the `strimzi-kafka-operator`'s system test module offers a streamlined, efficient approach to ensuring the performance standards of Strimzi-managed Kafka clusters are met and exceeded. 
```

File: 068-performance-testing.md (hunk 18 above)
Comment (see-quick) [reply 068-performance-testing.md]: Yeah, nice catch :)

File: 068-performance-testing.md (hunk 20 above)
Comment (see-quick) [reply 068-performance-testing.md]: done added.

File: 068-performance-testing.md (hunk 24 above)
Comment (see-quick) [reply 068-performance-testing.md]: Testing farm fixed

File: 068-performance-testing.md (hunk 30 above)
Comment (see-quick) [reply 068-performance-testing.md]: yeah good idea

File: 068-performance-testing.md (hunk 25 above)
Comment (see-quick) [reply 068-performance-testing.md]: Yeah, but this just shows what such tests would look like. It's just for illustrative purposes for a reader to have some clue about:
1. setup with also ThreadCollector ...
2. exercise 
3. verify
4. teardown - performance reporter...

I could delete it or maybe make it shorter

File: 068-performance-testing.md (hunk 17 above)
Comment (see-quick) [reply 068-performance-testing.md]: Removed
